File: frontend/src/app/models/chatdto.ts
import { UserDto } from "./userdto";
import { MessageDto } from "./messagedto";
import { UserChatDto } from "./userchatdto";

export class ChatDto {
  constructor(
    public id?: string,
    public chatType?: number,
    public userChats?: UserChatDto[],
    public messages?: MessageDto[],
    public creatorId?: string,
    public creator?: UserDto,
    public title?: string
    ) { }
}

File: backend/src/MessagingPlatform.Application/Common/Models/ChatDTOs/ChatDto.cs
using MessagingPlatform.Application.Common.Models.MessageDTOs;
using MessagingPlatform.Application.Common.Models.UserChatDTOs;
using MessagingPlatform.Application.Common.Models.UserDTOs;

namespace MessagingPlatform.Application.Common.Models.ChatDTOs;

public class ChatDto
{
    public Guid Id { get; set; }
    public int ChatType { get; set; }
    public ICollection<UserChatDto>? UserChats { get; set; }
    public ICollection<MessageDto>? Messages { get; set; }
    public string Title { get; set; } = "New Chat";
}
